### PR TITLE
op-supervisor: initial service draft

### LINKED ISSUE #10859: Interop: op-supervisor service skeleton setup

### PR DESCRIPTION
**Pull request by protolambda**
**Description**

Draft of the service-skeleton for standalone superchain-backend. 

See design-doc for context: https://github.com/ethereum-optimism/design-docs/pull/9/files

This does not yet implement the opinionated parts of the DB updates and safety query execution.

**Tests**

Spins up the service with mock backend and runs a test RPC query, then shuts down the service.

**Metadata**

Fix https://github.com/ethereum-optimism/optimism/issues/10859

### CODE REVIEW

File: op-supervisor/supervisor/cli_config.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+type CLIConfig struct {
+	Version string
+
+	LogConfig     oplog.CLIConfig
+	MetricsConfig opmetrics.CLIConfig
+	PprofConfig   oppprof.CLIConfig
+	RPC           oprpc.CLIConfig
+
+	// MockRun runs the service with a mock backend
+	MockRun bool
```
Comment (ajsutton) [inline op-supervisor/supervisor/cli_config.go:24]: Probably should set this based on the cli flag?

File: op-supervisor/supervisor/service.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	if err := su.pprofService.Start(); err != nil {
+		return fmt.Errorf("failed to start pprof service: %w", err)
+	}
+
+	return nil
+}
+
+func (su *SupervisorService) initMetricsServer(cfg *CLIConfig) error {
+	if !cfg.MetricsConfig.Enabled {
+		su.log.Info("metrics disabled")
```
Comment (ajsutton) [inline op-supervisor/supervisor/service.go]:
```suggestion
		su.log.Info("Metrics disabled")
```

Comment (ajsutton) [reply op-supervisor/supervisor/service.go]: I'm pushing a commit to fix all of the log capitalisation. Some hills are worth dying on. 🤣

**Review by ajsutton — APPROVED** [review]
LGTM but I think we should set it up so we test the CLI flags - that's been really successful in op-challenger and op-dispute-mon (see their `main_test.go` files) to avoid things like forgetting to set the value of a CLI flag or forgetting to add a flag to the list that actually gets used etc which we hit surprisingly regularly otherwise. Can do that as a follow up though.

File: op-supervisor/supervisor/cli_config.go (hunk 1 above)
Comment (protolambda) [reply op-supervisor/supervisor/cli_config.go:24]: good catch, fixed

**Pull request merged**